gtsam/geometry/Similarity2.h

```diff
@@ -0,0 +1,189 @@
+
+  /// @}
+  /// @name Testable
+  /// @{
+
+  /// Compare with tolerance
+  GTSAM_EXPORT bool equals(const Similarity2& sim, double tol) const;
+
+  /// Exact equality
+  GTSAM_EXPORT bool operator==(const Similarity2& other) const;
+
+  /// Print with optional string
+  GTSAM_EXPORT void print(const std::string& s) const;
+
+  GTSAM_EXPORT friend std::ostream &operator<<(std::ostream &os, const Similarity2& p);
+
+  /// @}
+  /// @name Group
+  /// @{
+
+  /// Return an identity transform
+  GTSAM_EXPORT static Similarity2 identity();
+
+  /// Composition
+  GTSAM_EXPORT Similarity2 operator*(const Similarity2& S) const;
+
+  /// Return the inverse
+  GTSAM_EXPORT Similarity2 inverse() const;
+
+  /// @}
+  /// @name Group action on Point2
+  /// @{
+
+  /// Action on a point p is s*(R*p+t)
+  GTSAM_EXPORT Point2 transformFrom(const Point2& p) const;
+
+  /** 
+   * Action on a pose T.
+   * |Rs  ts|   |R t|   |Rs*R Rs*t+ts| 
+   * |0  1/s| * |0 1| = | 0      1/s |, the result is still a Sim2 object.
+   * To retrieve a Pose2, we normalized the scale value into 1.
+   * |Rs*R Rs*t+ts|   |Rs*R s(Rs*t+ts)|
+   * | 0      1/s | = |  0       1    |
+   * 
+   * This group action satisfies the compatibility condition. 
+   * For more details, refer to: https://en.wikipedia.org/wiki/Group_action
+   */
+  GTSAM_EXPORT Pose2 transformFrom(const Pose2& T) const;
+
+  /** syntactic sugar for transformFrom */
+  GTSAM_EXPORT Point2 operator*(const Point2& p) const;
+
+  /**
+   *  Create Similarity2 by aligning at least two point pairs
+   */
+  GTSAM_EXPORT static Similarity2 Align(const std::vector<Point2Pair>& abPointPairs);
+  
+  /**
+   * Create the Similarity2 object that aligns at least two pose pairs.
+   * Each pair is of the form (aTi, bTi).
+   * Given a list of pairs in frame a, and a list of pairs in frame b, Align()
+   * will compute the best-fit Similarity2 aSb transformation to align them.
+   * First, the rotation aRb will be computed as the average (Karcher mean) of
+   * many estimates aRb (from each pair). Afterwards, the scale factor will be computed
+   * using the algorithm described here:
+   * http://www5.informatik.uni-erlangen.de/Forschung/Publikationen/2005/Zinsser05-PSR.pdf
+   */
+  GTSAM_EXPORT static Similarity2 Align(const std::vector<Pose2Pair>& abPosePairs);
+
+  /// @}
+  /// @name Lie Group
+  /// @{
+
+  using LieGroup<Similarity2, 4>::inverse;
+
+  /// @}
+  /// @name Standard interface
+  /// @{
+
+  /// Calculate 4*4 matrix group equivalent
+  GTSAM_EXPORT const Matrix3 matrix() const;
+
+  /// Return a GTSAM rotation
+  const Rot2& rotation() const {
+    return R_;
+  }
+
+  /// Return a GTSAM translation
+  const Point2& translation() const {
+    return t_;
+  }
+
+  /// Return the scale
+  double scale() const {
+    return s_;
+  }
+
+  /// Convert to a rigid body pose (R, s*t)
+  /// TODO(frank): why is this here? Red flag! Definitely don't have it as a cast.
+  GTSAM_EXPORT operator Pose2() const;
```

Comment:
Interesting that I have comments in this new PR ?? Must have forgotten. In any case, comment holds: remove this cast operator, as I don’t think there is a valid way to cast down, only cast from Pose2 to Sim2.

Reply:
This is actually copy-pasted from `Similarity3.h`. We should probably address that there.